jupyter_console/ptshell.py

```diff
@@ -708,22 +708,22 @@ def handle_iopub(self, msg_id=''):
     def handle_rich_data(self, data):
         for mime in self.mime_preference:
             if mime in data and mime in self._imagemime:
-                self.handle_image(data, mime)
-                return True
+                return self.handle_image(data, mime)
 
     def handle_image(self, data, mime):
         handler = getattr(
             self, 'handle_image_{0}'.format(self.image_handler), None)
         if handler:
-            handler(data, mime)
+            return handler(data, mime)
 
     def handle_image_PIL(self, data, mime):
         if mime not in ('image/png', 'image/jpeg'):
-            return
+            return False
         import PIL.Image
         raw = base64.decodestring(data[mime].encode('ascii'))
         img = PIL.Image.open(BytesIO(raw))
         img.show()
```

Comment:
well, [this](https://github.com/python-pillow/Pillow/blob/3.2.0/PIL/ImageShow.py#L49-L53) is the `show` i use in my PR.

as you can see [later](https://github.com/python-pillow/Pillow/blob/3.2.0/PIL/ImageShow.py#L162-L175), it’s possible that neither `xv` nor `display` are installed, which makes your indirect show function fail silently.